mne/label.py

```diff
@@ -2020,6 +2036,150 @@ def read_labels_from_annot(subject, parc='aparc', hemi='both',
+    mmaps = dict((hemi, mmap) for hemi, mmap in zip(('lh', 'rh'), mmaps))
+    vert_poss = dict((hemi, _load_vert_pos(
+        subject_to, subjects_dir, surf_name, hemi, mmap.shape[0]))
+        for hemi, mmap in mmaps.items())
+    mmaps = dict((hemi, mmap.argmax(-1)) for hemi, mmap in mmaps.items())
+    out_labels = list()
+    values = filename = None
+    for label in labels:
+        vertices = np.where(np.in1d(mmaps[label.hemi], label.vertices))[0]
+        pos = vert_poss[label.hemi][vertices]
+        out_labels.append(
+            Label(vertices, pos, values, label.hemi, label.comment, label.name,
+                  filename, subject_to, label.color, label.verbose))
+    return out_labels
+
+
+@verbose
+def labels_to_stc(labels, values, tmin=0, tstep=1, subject=None, verbose=None):
+    """Convert a set of labels and values to a STC.
+
+    Parameters
+    ----------
+    labels : list of Label
+        The labels. Must not overlap.
+    values : ndarray, shape (len(labels), ...)
+        The values in each label. Can be 1D or 2D.
+    tmin : float
+        The tmin to use for the STC.
+    tstep : float
+        The tstep to use for the STC.
+    subject : str | None
+        The subject for which to create the STC.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    stc : instance of SourceEstimate
+        The values-in-labels converted to a STC.
+
+    Notes
+    -----
+    Vertices that appear in more than one label will be averaged.
+
+    .. versionadded:: 0.18
+    """
+    subject = _check_labels_subject(labels, subject, 'subject')
+    values = np.array(values, float)
+    if values.ndim == 1:
+        values = values[:, np.newaxis]
+    if values.ndim != 2:
+        raise ValueError('values must have 1 or 2 dimensions, got %s'
+                         % (values.ndim,))
+    if len(labels) != len(values):
+        raise ValueError('values.shape[0] (%s) must match len(labels) (%s)'
+                         % (values.shape[0], len(labels)))
+    vertices = dict(lh=[], rh=[])
+    data = dict(lh=[], rh=[])
+    for li, label in enumerate(labels):
+        data[label.hemi].append(
+            np.repeat(values[li][np.newaxis], len(label.vertices), axis=0))
+        vertices[label.hemi].append(label.vertices)
+    hemis = ('lh', 'rh')
+    for hemi in hemis:
+        vertices[hemi] = np.concatenate(vertices[hemi], axis=0)
+        data[hemi] = np.concatenate(data[hemi], axis=0).astype(float)
+        cols = np.arange(len(vertices[hemi]))
+        vertices[hemi], rows = np.unique(vertices[hemi], return_inverse=True)
+        mat = sparse.coo_matrix((np.ones(len(rows)), (rows, cols))).tocsr()
+        mat = mat * sparse.diags(1. / np.asarray(mat.sum(axis=-1))[:, 0])
+        data[hemi] = mat.dot(data[hemi])
+    vertices = [vertices[hemi] for hemi in hemis]
```

Comment:
I don't understand why we first process them by hemis if we end up flattening them.

Reply:
labels are organized per-hemisphere (their vertices refer to vertices in given hemi). But stc assumes data that is not split per hemi and a list of hemi-split vertices.